### ISSUE TITLE
Update R&S prototype based on conversation guide

### ISSUE THREAD

**Issue by davidmpickett** [department-of-veterans-affairs/va.gov-cms#20075]
## Dependencies

**This ticket is blocked by:**
- [x] #19705 
- [x] Part of #20002

**This ticket blocks**
- [ ] #19703 

## Description

In #19000, we built a functional prototype that performs all the logic necessary for a "good enough" search and filtering experience for research purposes. However, some content in the prototype was left blank because it is dependent on the particulars of the Conversation Guide. In other words the prototype was essentially feature complete, but not content complete.

This ticket will cover filling in the missing pieces of the prototype and get it ready for research.

## Landing page / search interface
- [x] Search interface - prompts and related logic for results screen (dependent on [conversation guide](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/19705))
- [x] ~~Featured - third link should be canned search~~
   - This was buggy and not needed for conversation guide. I reverted it to going directly to an article detail page.

## Results page / filter drawer
- [x] Results update based on search prompt (dependent on [conversation guide](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/19705))
- [x] Allow user to change search prompt while on this page?
- [x] Update "Apply Filter" button to account for search logic
- [x] Results count message - "Showing 1 - X of Y results" starts at a high number and gets lower when search term or filters applied
- [x] Make sure all 30 links have correct custom logic settings for what to show on detail page (10 were coded in 19000, but all will need to be redone since base logic is changing) 
- [x] Update the X (close icon) in filter drawer to have same logic as "apply filters" button. Makes it functional even if not final intended functionality
- [x] Update breadcrumb link to reset all variables
- [x] Dismissing a Filter chip updates results (big lift)
- [x] Clicking Clear All updates results

## Detail pages
- [x] Mobile layout (may be completed by [design ticket](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/20002))
- [x] Confirm all 30 pages update correctly when clicking on search result
- [x] Update links from filter chips to clear out other variables before loading results page again
- [x] Update breadcrumb link to reset all variables

## Confirm out of scope
- [x] Updating filter chips to have border
- [x] Featured - third link should be canned search

**Comment by jilladams** [comment]
This ticket will run in parallel to Cindy updating the conversation guide, so some work can happen now and some will happen after convo guide updates. It's likely to span sprints 21 & 22.

**Comment by davidmpickett** [comment]
@cindymerrill I am copying your question from Slack over here since this will be the ticket where I adjust Prototype functionality. 

> I have a design/prototype question:
> If you click a common topic or benefit from the landing page, the prototype shows the article along with filters you didn't select. Seems like the filters shouldn't be shown because the user didn't select them, or maybe this was purposeful to show that filters exist?
> 
>![Image](https://github.com/user-attachments/assets/be919373-8f0c-4eaf-9aec-ffc0394f104b)

**Intended behavior**
- A user selects a common topic or benefit from the landing page (e.g. Claims and claim status)
- The user is taken to the results page with the corresponding filter applied

Landing page | Results page
-- | --
![Image](https://github.com/user-attachments/assets/59daa5ae-a634-4173-8528-51c35440a593) | ![Image](https://github.com/user-attachments/assets/86e76785-7753-4fe1-83b6-a6ff11dfe8d6)

**Possible bug**
The screenshot you shared showing two filters applied after clicking a link from the landing page should not be possible. However there are flaky aspects to the prototype, so it's possible the sequence of inputs you used circumvented the systems I built. 

If you are able to reproduce this issue and document the exact series of steps you take to get multiple filters to show, that would help me understand where my logic might have a hole in it

**Comment by cindymerrill** [comment]
Makes sense @davidmpickett. I'll let you know if I reproduce this issue again...

**Comment by davidmpickett** [comment]
Added mobile layout of detail page to the prototype - should be a lot easier to use and understand the whole flow now

**Comment by cindymerrill** [comment]
@davidmpickett I think you've left for the day, and I'm having trouble getting the prototype to work. I know you're still working on it, but I just wanted to document what I noticed so far...

1. The search button doesn't work. I was able to run searches earlier, but not now. [Time passes] Now it's working! Did you fix something, Dave? If not, then we should figure out why it sometimes works and other times doesn't. 
2. Community care in Common topics returns only one result. I thought there would be more.
3. Some of the Common topics show content that seems totally irrelevant (e.g., all results for Burials and memorials, all results for Careers and employment, "Career resources for military and surviving spouses" for Health care).
4. Sometimes when I try to get back to the landing page using the breadcrumb, a filter that I didn't use appears. Sorry I can't reproduce the steps that preceded this, so I'll watch to see if it happens again.

I'll stop here and wait to hear from you before I test more.

**Comment by davidmpickett** [comment]
> The search button doesn't work. I was able to run searches earlier, but not now. [Time passes] Now it's working! Did you fix something, Dave? If not, then we should figure out why it sometimes works and other times doesn't.

Yup, that was me. I accidentally broke it while trying to do something else and then fixed it. I'm now signing off for the day, so it will be in a stable state until Monday :)

**Comment by cindymerrill** [comment]
FYI I'm not going to be any more testing until Monday. See you then...

**Comment by davidmpickett** [comment]
> 2. Community care in Common topics returns only one result. I thought there would be more.

This is functioning as intended. Currently there is only one article/result tagged as Community Care. See the [spreadsheet](https://docs.google.com/spreadsheets/d/1-4tl2sUeBUSqJcWvjdx6_8rA1o2xOaU7pitDPad_Tk0/edit?pli=1&gid=2058982768#gid=2058982768)-> Detail Pages tab -> Topics column.

We do have 4 unused results, so we make 1 or 2 of those Community Care related.

> 3. Some of the Common topics show content that seems totally irrelevant (e.g., all results for Burials and memorials, all results for Careers and employment, "Career resources for military and surviving spouses" for Health care).

"Common topics" and "Browse by benefit" links on the landing page are currently broken. They are using the old logic for the original 30 articles. I will work on updating them to the new logic and let you know when they are fixed.

> 4. Sometimes when I try to get back to the landing page using the breadcrumb, a filter that I didn't use appears. Sorry I can't reproduce the steps that preceded this, so I'll watch to see if it happens again.> > I'll stop here and wait to hear from you before I test more.

I still need to revisit the "reset" logic for the breadcrumb. Once I've addressed it again, I'll let you know so we can see if this is still happening

**Comment by davidmpickett** [comment]
Current To Do list
- [ ] Add 2 Community care results ( use 2 empty slots)
- [ ] Make sure all filters return at least 1 result (even if not super relevant)
- [ ] Make sure all filters and searches expected in tasks return at least 3 results
- [ ] Fix "Browse by benefits" link
- [ ] Update Evidence result to show under Disability
- [ ] Update logic that resets variables when using breadcrumb
- [ ] Tags on detail pages clickable

Up for conversation
- Dismissible filters - have to click search after

**Comment by davidmpickett** [comment]
**End of sprint update**
Still working on this, just got a final list of content yesterday, need to update that into prototype. Also trying to resolve some bugs and figure out why the prototype isn't working on some device/browser configurations

**Comment by cindymerrill** [comment]
I'm planning to launch recruiting today. If we need to restrict mobile device and/or browser combinations for research participants in order to have a working prototype, I'll need to know ASAP so I can update the recruiting criteria. @davidmpickett @rlittleCivicActions

**Comment by davidmpickett** [comment]
Just confirmed that the prototype matches basic spreadsheet logic for the two primary interaction paths

**Applying 1 filter at a time via filter drawer and clicking apply filters**
- Veterans = correctly returns 29 results
- Family members and caregivers = correctly returns 27 results
- Active service members = correctly returns 21 results
- Burial and memorial = correctly returns 5 results
- Careers and employment = correctly returns 3 results
- Disability = correctly returns 10 results
- Education and training = correctly returns 8 results
- Health care = correctly returns 11 results
- Home loans and housing assistance = correctly returns 2 results
- Life insurance = correctly returns 2 results
- Pension = correctly returns 8 results
- Survivor compensation = correctly returns 6 results
- Claims and claim status = correctly returns 3 results
- Community care = correctly returns 3 results
- Decision reviews and appeals = correctly returns 3 results
- Payments and debt = correctly returns 9 results
- Records and documents = correctly returns 3 results
- Sign in and profile = correctly returns 3 results

**Choosing a search term from landing page and clicking search button**
- C "pay health care" = correctly returns 10 results
- D "reconsider benefits" = correctly returns 10 results
- E "application documents"	= correctly returns 7 results
- J "job help" =  correctly returns 4 results
- P "owe VA money" = correctly returns 13 results
- S "get into VA account" = correctly returns 17 results	
- T "submit train receipt" = correctly returns 5 results

Still need to fix the links on the landing page and clicking the tag chips on detail pages

**Comment by davidmpickett** [comment]
Results 1-26 show the correct tags on their detail pages.
- [ ] Need to code 27-30
- [ ] Need to double check featured links

**Comment by davidmpickett** [comment]
Seems basically done! Will keep ticket open until pilot sessions are complete in case anything else pops up

**Comment by cindymerrill** [comment]
Wow, @davidmpickett , the prototype looks great and has so much content in it that comes up when it's supposed to!! I'm excited to have the first pilot participant try it out tomorrow...

I just tested everything on my phone, which didn't work at first, though it worked after I closed the browser window and tried again. We might need to advise participants to do this if the prototype seems unresponsive *after they've waited a while first*. I found that some of the interactions seem really slow, especially "Apply filters" and "Clear all"--perhaps because of all the code/variables doing the hard work behind the scenes?

**Comment by davidmpickett** [comment]
Attaching excel version of the content spreadsheet in case it is ever needed for future reference

[Content needed for R&S search enhancements research 2_2025.xlsx](https://github.com/user-attachments/files/18697429/Content.needed.for.R.S.search.enhancements.research.2_2025.xlsx)

**Comment by davidmpickett** [comment]
Following up on the third pilot, I made some final tweaks to how scroll position works. I have tested all the various scenarios for scrolling and I think they are all as good as can be now in Figma. Closing this ticket
